global_shortcuts.go

```diff
@@ -0,0 +1,135 @@
+		return
+	})
+	return
+}
+
+// Register Register global shortcuts
+func (gs *GlobalShortcuts) Register(accelerator string, callback globalShortcutsCallback) (isRegistered bool, err error) {
+
+	if err = gs.ctx.Err(); err != nil {
+		return
+	}
+
+	// Send an event to astilectron to register the global shortcut
+	var event = Event{Name: EventNameGlobalShortcutsCmdRegister, TargetID: gs.id, GlobalShortcuts: &EventGlobalShortcuts{Accelerator: accelerator}}
+	result, err := synchronousEvent(gs.ctx, gs, gs.w, event, EventNameGlobalShortcutsEventRegistered)
+
+	if err != nil {
+		return
+	}
+
+	// If registered successfully, add the callback to the map
+	if result.GlobalShortcuts.IsRegistered {
+		gs.m.Lock()
+		gs.callbacks[accelerator] = &callback
+		gs.m.Unlock()
+	}
+
+	isRegistered = result.GlobalShortcuts.IsRegistered
+	return
+}
+
+// IsRegistered Check if a global shortcut is registered
+func (gs *GlobalShortcuts) IsRegistered(accelerator string) (isRegistered bool, err error) {
+
+	if err = gs.ctx.Err(); err != nil {
+		return
+	}
+
+	// Send an event to astilectron to check if global shortcut is registered
+	var event = Event{Name: EventNameGlobalShortcutsCmdIsRegistered, TargetID: gs.id, GlobalShortcuts: &EventGlobalShortcuts{Accelerator: accelerator}}
+	result, err := synchronousEvent(gs.ctx, gs, gs.w, event, EventNameGlobalShortcutsEventIsRegistered)
+
+	if err != nil {
+		return
+	}
+
+	isRegistered = result.GlobalShortcuts.IsRegistered
+	return
+}
+
+// Unregister Unregister a global shortcut
+func (gs *GlobalShortcuts) Unregister(accelerator string) (err error) {
+
+	if err = gs.ctx.Err(); err != nil {
+		return
+	}
+
+	// Send an event to astilectron to unregister the global shortcut
+	var event = Event{Name: EventNameGlobalShortcutsCmdUnregister, TargetID: gs.id, GlobalShortcuts: &EventGlobalShortcuts{Accelerator: accelerator}}
+	_, err = synchronousEvent(gs.ctx, gs, gs.w, event, EventNameGlobalShortcutsEventUnregistered)
+
+	if err != nil {
+		return
+	}
+
+	// No need to find the callback from the map and delete it
+	//  because that event will no longer be triggerred
+	// If the same global shortcut is registered again, the original callback will be replaced with the new one
+
+	return
+}
+
+// UnregisterAll Unregister all global shortcuts
+func (gs *GlobalShortcuts) UnregisterAll() (err error) {
+
+	if err = gs.ctx.Err(); err != nil {
+		return
+	}
+
+	// Send an event to astilectron to unregister all global shortcuts
+	var event = Event{Name: EventNameGlobalShortcutsCmdUnregisterAll, TargetID: gs.id}
+	_, err = synchronousEvent(gs.ctx, gs, gs.w, event, EventNameGlobalShortcutsEventUnregisteredAll)
+
+	if err != nil {
+		return
+	}
+
+	gs.m.Lock()
+	gs.callbacks = make(map[string]*globalShortcutsCallback) // Clear the map
+	gs.m.Unlock()
+
+	return
+}
+
+// execCallback Execute the GlobalShortcuts event triggered from astilectron
+func (gs *GlobalShortcuts) execCallback(accelerator string) {
```

Comment:
I don't think there's a need for this function since it's called only once and it's not that big 🤔